Effects/DynamicShadows/source/PlayState.hx

```diff
@@ -0,0 +1,261 @@
+	 */
+	private var lineStyle:LineStyle;
+	
+	/**
+	 * Used for drawing stuff
+	 */
+	private var drawStyle:DrawStyle;
+	
+	private var infoText:FlxText;
+	
+	override public function create():Void
+	{
+		super.create();
+		
+		random = new FlxRandom();
+		FlxG.camera.bgColor = 0x5a81ad;
+		lineStyle = { color:shadowColor, thickness:1 };
+		drawStyle = {matrix:new Matrix(), colorTransform:new ColorTransform(1,1,1)};
+		
+		FlxNapeSpace.init();
+		FlxNapeSpace.space.gravity.setxy(0, 1200);
+		FlxNapeSpace.drawDebug = false; // You can toggle this on/off one by pressing 'D'
+		
+		var background:FlxTilemap = new FlxTilemap();
+		background.loadMapFromCSV("assets/data/background.txt", "assets/images/tiles.png", TILE_SIZE, TILE_SIZE, null, 1, 1);
+		add(background);
+		
+		// Note: The tilemap used in this demo was drawn with 'Tiled' (http://www.mapeditor.org/), but the level data was picked from the .tmx file and put into two separate
+		// .txt files for simplicity. If you wish to learn how to use Tiled with your project, have a look at this demo: http://haxeflixel.com/demos/TiledEditor/
+		
+		// If we add the shadows *before* all of the foreground elements (stage included) they will only cover the background, which is usually what you'd want I'd guess :)
+		shadowCanvas = new FlxSprite();
+		shadowCanvas.blend = BlendMode.MULTIPLY;
+		shadowCanvas.makeGraphic(FlxG.width, FlxG.height, FlxColor.TRANSPARENT, true);
+		add(shadowCanvas);
+		
+		foreground = new FlxNapeTilemap();
+		foreground.loadMapFromCSV("assets/data/foreground.txt", "assets/images/tiles.png", TILE_SIZE, TILE_SIZE, null, 1, 1);
+		add(foreground);
+		
+		foreground.setupTileIndices([4]);
+		
+		var tileIndex:Int = 0;
+		for (tileY in 0...foreground.heightInTiles)
+		{
+			for (tileX in 0...foreground.widthInTiles)
+			{
+				tileIndex = foreground.getTile(tileX, tileY);
+				var xPos:Float = tileX * TILE_SIZE;
+				var yPos:Float = tileY * TILE_SIZE;
+				
+				if (tileIndex == PROP_BARREL)
+				{					
+					add(new Barrel(xPos, yPos));
+					cleanTile(tileX, tileY);
+				}
+				else if (tileIndex == PROP_GEM)
+				{
+					gem = new Gem(xPos, yPos);
+					add(gem);
+					cleanTile(tileX, tileY);
+				}
+			}
+		}
+		
+		shadowOverlay = new FlxSprite();
+		shadowOverlay.makeGraphic(FlxG.width, FlxG.height, FlxColor.TRANSPARENT, true);
+		shadowOverlay.blend = BlendMode.MULTIPLY;
+		add(shadowOverlay);
+		
+		infoText = new FlxText(10, 10, 100, "");
+		add(infoText);
+	}
+	
+	/**
+	 * The tile in question was replaced by an actual object, this function will clean up the tile for us
+	 */
+	private function cleanTile(_x:Int, _y:Int):Void
+	{
+		foreground.setTile(_x, _y, 0);
+	}
+	
+	override public function update(elapsed:Float):Void
+	{
+		infoText.text = "FPS: " +Std.string(FlxG.game.debugger.stats.currentFps()).substr(0, 4) +"\n\nObjects can be dragged/thrown around.\n\nPress 'R' to restart.";
```

Comment:
iirc, I couldn't find a way to extract the current fps from the openfl-fps, I could try again, or just remove it altogether if I can't make it work. but yes, I wanted the fonts of the fps and info to match, so that's what I ended up doing.